Sources/AppCoinsSDK/Domain/Entities/Purchase.swift

```diff
@@ -162,8 +173,8 @@ public class Purchase: Codable {
                 
                 group.notify(queue: .main) {
                     if isConsumed { continuation.resume() }
-                    else if isNetworkError { continuation.resume(throwing: AppCoinsSDKError.networkError) }
-                    else { continuation.resume(throwing: AppCoinsSDKError.unknown) }
+                    else if isNetworkError, let error = error { continuation.resume(throwing: error) }
+                    else { continuation.resume(throwing: AppCoinsSDKError.unknown(message: "Failed to complete the purchase process", description: "The purchase was not consumed and the item was not attributed to the user")) }
```

Comment:
Missing code location on the description.

Reply:
Done